model/modeldecoder/modeldecodertest/populator.go

```diff
@@ -119,14 +252,50 @@ func iterateStruct(v reflect.Value, key string, fn func(f reflect.Value, fKey st
 		}
 		fKey = fmt.Sprintf("%s%s", key, name)
 
-		if fTyp.Kind() == reflect.Struct {
+		// call the given function with every field
+		fn(f, fKey)
+		// check field type for recursive iteration
+		switch f.Kind() {
+		case reflect.Ptr:
+			if !f.IsZero() && fTyp.Elem().Kind() == reflect.Struct {
+				iterateStruct(f.Elem(), fKey, fn)
+			}
+		case reflect.Struct:
 			switch f.Interface().(type) {
-			case nullable.String, nullable.Int, nullable.Interface:
+			case nullable.String, nullable.Int, nullable.Bool, nullable.Float64,
+				nullable.Interface, nullable.HTTPHeader, nullable.TimeMicrosUnix:
 			default:
 				iterateStruct(f, fKey, fn)
 			}
+		case reflect.Map:
+			l := f.Len()
+			if l == 0 {
+				continue
+			}
+			// values in maps are not adressable, therefore we need a workaround:
+			// adding the value to a temporary slice, passing this value into the iterateStruct
+			// function call and after it was potentially modified, setting it as new value
+			// in the map
+			tmpSlice := reflect.MakeSlice(reflect.SliceOf(f.Type().Elem()), l, l)
+			var i int
+			for _, mKey := range f.MapKeys() {
+				mapVal := f.MapIndex(mKey)
+				tmpSlice.Index(i).Set(mapVal)
+				if mapVal.Kind() == reflect.Struct {
+					sVal := tmpSlice.Index(i)
+					iterateStruct(sVal, fmt.Sprintf("%s.[%s]", fKey, mKey), fn)
+					f.SetMapIndex(mKey, sVal)
+				}
+				i++
+			}
```

Comment:
AFAICS, you don't need a slice here. Use `reflect.New` to create an addressable value instead:

```suggestion
			iter := f.MapRange()
			for iter.Next() {
				mKey := iter.Key()
				mVal := iter.Value()
				ptr := reflect.New(mVal.Type())
				ptr.Elem().Set(mVal)
				iterateStruct(ptr.Elem(), fmt.Sprintf("%s.[%s]", fKey, mKey), fn)
				f.SetMapIndex(mKey, ptr.Elem())
			}
```

Reply:
The `iterateStruct` should not change anything - iterate is called from the `set` methods, but also from `assert` - it should not change the structure but only iterate through it and allow for the structure to be changed when calling the `fn` function.

Reply:
I'm afraid I don't follow. My suggestion is only to stop using a temporary slice, and instead use a temporary variable (reflect.New). If I glossed over something though, ignore me for now and we can discuss post-merge.